README.md

```diff
@@ -230,6 +230,44 @@ jobs:
 ```
 </details>
 
+## Deleting caches
+We can not re-use caches from pull request branches in other branches like main, such caches can eat up the storage quota and hence causing thrashing on more useful branches like main. In order to resolve this issue, we can use [gh-actions-cache cli](https://github.com/actions/gh-actions-cache/) to delete caches. This workflow uses `gh-actions-cache` to delete all the caches created by all the pull requests. 
+
+```
+name: cleanup-caches
+on:
+  schedule:
+    - cron: '0 */3 * * *'
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Cleanup
+        run: |
+          gh extension install actions/gh-actions-cache
+          
+          REPO=${{ github.repository }}
+
+          echo "Fetching list of cache key"
+          ## This will extract out all the cache keys for pull requests
+          cacheKeysForPR=$(gh actions-cache list -R $REPO | grep "refs/pull" | cut -d $'\t'  -f 1 )
```

Comment:
So this will list caches generated by all PR branches, let's say active PRs too? Which means active PRs will have to build their program from scratch every time this job runs.

Instead should we try something like deleting cache only for the PR that is closed or merged?

Reply:
The workflow is a sample workflow to provide information on how to delete caches using gh-actions-cache, users are free to modify this based on their specific case.